app/uk/gov/hmrc/pillar2/models/audit/CreateSubscriptionAuditEvent.scala

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.pillar2.models.audit
+
+import play.api.libs.json.{JsValue, Json, OFormat, OWrites}
+import uk.gov.hmrc.pillar2.models.hods.RegisterWithoutIDRequest
+import uk.gov.hmrc.pillar2.models.hods.subscription.request.RequestDetail
+
+case class CreateSubscriptionAuditEvent(
+  requestData:  RequestDetail,
+  responseData: AuditResponseReceived
+) extends AuditEvent {
+  override val auditType:  String  = "CreateP2Subscription"
+  override val detailJson: JsValue = Json.toJson(this)
+}
+
+object CreateSubscriptionAuditEvent {
+  implicit val format: OFormat[CreateSubscriptionAuditEvent] = Json.format[CreateSubscriptionAuditEvent]
+  implicit val writes: OWrites[CreateSubscriptionAuditEvent] = Json.writes[CreateSubscriptionAuditEvent]
+}
```

Comment:
I think the OFormat and OWrite can be combined as a single formats, but happy to go with yours.
See the following:
https://github.com/hmrc/trusts/blob/main/app/models/auditing/TrustRegistrationSubmissionAuditEvent.scala#L26